internal/reflect/pointer_test.go

```diff
@@ -101,3 +102,31 @@ func TestFromPointer_null(t *testing.T) {
 		t.Errorf("Unexpected diff (+wanted, -got): %s", diff)
 	}
 }
+
+func TestFromPointer_AttrTypeWithValidate_Error(t *testing.T) {
+	v := "hello, world"
+	_, diags := refl.FromPointer(context.Background(), testtypes.StringTypeWithValidateError{}, reflect.ValueOf(&v), tftypes.NewAttributePath())
+	if len(diags) == 0 {
+		t.Fatalf("expected diagnostics, got none")
+	}
+	if !reflect.DeepEqual(diags[0], testtypes.TestErrorDiagnostic) {
```

Comment:
I think so far we've been using `cmp.Equal` for these. Does that not work here? I think it's a bit more flexible than `reflect.DeepEqual`.

Reply:
Sorry not used to that yet, will refactor.

Reply:
Updated in 28696c08696c746a258bd590b55481bfd2dfccad